src/veritysetup/veritysetup.c

```diff
@@ -109,11 +121,63 @@ static int parse_options(const char *options) {
                 else if (streq(word, "panic-on-corruption"))
                         arg_activate_flags |= CRYPT_ACTIVATE_PANIC_ON_CORRUPTION;
 #endif
-                else if ((val = startswith(word, "hash-offset="))) {
+                else if ((val = startswith(word, "no-superblock="))) {
+
+                        r = parse_boolean(val);
+                        if (r < 0)
+                                log_error_errno(r, "Failed to parse %s, ignoring: %m", word);
+
+                        arg_no_superblock = r;
+                } else if ((val = startswith(word, "format="))) {
+
+                        r = safe_atoi(val, &arg_format);
+                        if (r < 0)
+                                log_error_errno(r, "Failed to parse %s, ignoring: %m", word);
+                } else if ((val = startswith(word, "data-block-size="))) {
+
+                        r = safe_atou64(val, &arg_data_block_size);
+                        if (r < 0)
+                                log_error_errno(r, "Failed to parse %s, ignoring: %m", word);
```

Comment:
i figure block sizes of zero or so should be refused here too (also below)

Reply:
I have introduced the function `block_size_is_valid()` which checks the value the same way as [cryptsetup](https://gitlab.com/cryptsetup/cryptsetup/-/blob/62f334cfa58f9a21b24facdec35fc99e4cb1a4a0/lib/verity/verity.h#L27-28).

I guess I should also check for the `*-offset` parameters as well.